.dagger/build/sdk.go

```diff
@@ -128,18 +129,22 @@ func (build *Builder) typescriptSDKContent(ctx context.Context) (*sdkContent, er
 }
 
 func (build *Builder) goSDKContent(ctx context.Context) (*sdkContent, error) {
-	base := dag.Container(dagger.ContainerOpts{Platform: build.platform}).
+	base := dag.Container().
 		From(consts.GolangImage).
-		WithExec([]string{"apk", "add", "git"})
+		With(build.goPlatformEnv).
+		WithEnvVariable("BUILDPLATFORM", "linux/"+runtime.GOARCH).
+		WithEnvVariable("TARGETPLATFORM", string(build.platform)).
+		WithExec([]string{"apk", "add", "git"}).
+		WithDirectory("/", dag.Container().From(consts.XxImage).Rootfs())
```

Comment:
This might be annoying or nonfeasible, but it's not ideal that we're including a ton of extra stuff in this go base image we include in the engine image. If in the practice the size difference before and after is negligible then doesn't matter. But if there is a meaningful size bump we could try to wipe this extra stuff out.

One possibility would be to run all this and then start with a clean `dag.Container().From(consts.GolangImage)` and copy in the stuff we added (mainly the mod download cache and the go build cache). You might also be able to do some fancy tricks with Diff+Merge (merge-op is triggered with copys that only involve `/` and no include/exclude), but that's a bit trickier since we hide merge-op as an implementation detail.

Reply:
Did this, since we should also be setting the `Platform` correctly - otherwise, we're building an image for the wrong platform entirely :scream: :scream: